File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
using namespace std;
const int MAX = 10000;
int q[MAX]{};
int head, tail = 0;
int main(void) {
	cin.tie(0);
	ios::sync_with_stdio(0);
	int N;
	cin >> N;
	for (int i = 0; i < N; i++) {
		string cmd; int n;
		cin >> cmd;
		if (cmd == "push") {
			cin >> n;
			q[tail++] = n;
		}
		else if (cmd == "size") {
			cout << tail - head << '\n';
		}
		else if (cmd == "empty") {
			if (tail - head == 0) {
				cout << 1 << '\n';
			} else { cout << 0  << '\n'; }
		}
		else if (cmd == "front") {
			if (tail - head == 0) {
				cout << -1 << '\n';
			}
			else {
				cout << q[head] << '\n';
			}
		}
		else if (cmd == "back") {
			if (tail - head == 0) 
				cout << -1 << '\n';
			else
			cout << q[tail - 1] << '\n';
		}
		else if (cmd == "pop") {
			if (head != tail) {
				cout << q[head++] << '\n';
			}
			else { cout << -1 << '\n'; }
		}
	}
}

File: 백준/Silver/10845. 큐/큐.py
## 10845
import sys
input = sys.stdin.readline
q = [0]*10000
first = 0
last = 0
N = int(input())
for i in range(N):
    cmd = input().split()
    if(cmd[0] == "push"):
        q[last] = cmd[1]
        last+=1
    elif(cmd[0] == "pop"):
        if(last-first == 0):
            print(-1)
        else:
            print(q[first])
            first+=1
    elif(cmd[0] == "size"):
        print(last-first)
    elif(cmd[0] == "empty"):
        if(last-first == 0):
            print(1)
        else:
            print(0)
    elif(cmd[0] == "front"):
        if(last-first == 0):
            print(-1)
        else:
            print(q[first])
    else:
        if(last-first == 0):
            print(-1)
        else:
            print(q[last-1])